### ISSUE TITLE
include `formatLocale` and `timeFormatLocale` parameters in options

### ISSUE THREAD

**Issue by mattijn** [vega/vega-embed#170]
Reasoning: I would like to set the `formatLocale` and `formatTimeLocale` as part of the `embed_options` in Altair.

At the vega repo, @jheer did comment once that you will need to invoke the `formatLocale` and `formatTimeLocale` methods _prior_ instantiation of a Vega View (https://github.com/vega/vega/pull/313#issuecomment-297562302).

Using this approach I can set it in VegaEmbed: https://observablehq.com/@mattijn/usage-of-timeformatlocale-formatlocale-in-vega-embed, but this is prior the invokation of the View constructor.

Basically its just:
```javascript
embed = require("vega-embed@4")
formatlocale = await (await fetch("https://unpkg.com/d3-format@1/locale/nl-NL.json")).json()
timeformatlocale = await (await fetch("https://unpkg.com/d3-time-format@1/locale/nl-NL.json")).json()
embed.vega.timeFormatLocale(timeformatlocale)
embed.vega.formatLocale(formatlocale)

viewof view = embed({
  "$schema": "https://vega.github.io/schema/vega-lite/v3.json",
  "description": "Google's stock price over time.",
  "width": 700,
  "data": {"url": "https://raw.githubusercontent.com/vega/vega-datasets/master/data/stocks.csv"},
  "transform": [{"filter": "datum.symbol==='GOOG'"}],
  "mark": {
    "type": "area",
    "color": "lightblue",
    "interpolate": "step-after",
    "line": true
  },
  "encoding": {
    "x": {"field": "date", "type": "temporal"},
    "y": {"field": "price", "type": "quantitative"},
    "tooltip": [
      {"type": "quantitative", "field": "price", "format": "$,"},
      {"type": "temporal", "field": "date"}
    ],
  },
  "selection": {
    "zoom_x": {"type": "interval", "bind": "scales", "encodings": ["x"]}
  },
})
```

Would it be possible to provide these locale settings as options in the Vega-Embed View?

Other sources of similar requests for Altair: 
- https://vega-js.slack.com/archives/C7K00SK98/p1560523113006600
- https://github.com/altair-viz/altair/issues/505

**Comment by domoritz** [comment]
I think adding a `locale` property to the embed options makes sense. By default, we should use the browser's locale, though. The question is how we can access d3 in the imported vega package since we are using vega as a peer dependency. Would it make sense to have an option in Vega to set the locale? Something like `vega.locale('nl-NL')`.

**Comment by sahrmann** [comment]
Would be great to have such a possibility. At the moment, it seems like saving the figures as HTML and changing the Locale there is the only way to make diagrams with different locale.

**Comment by domoritz** [comment]
Do you mean in Altair or observable? In the latter, you don’t need to save as html first.

**Comment by sahrmann** [comment]
I mean in Altair (using it in Jupyter Notebook). I tried to fork the ipyvega extension and put the formatLocale call in different places, but it does not change the output (see [https://github.com/sahrmann/ipyvega/blob/master/src/index.ts](https://github.com/sahrmann/ipyvega/blob/master/src/index.ts)).

**Comment by domoritz** [comment]
I'm somewhat hesitant to add this to Vega-Embed since formatting is a global property and not set per view. 

@jheer do you have a suggestion for how we should handle formatting in Vega-Embed and Altair/Jupyter?

**Comment by victornoel** [comment]
> I think adding a `locale` property to the embed options makes sense. By default, we should use the browser's locale, though. The question is how we can access d3 in the imported vega package since we are using vega as a peer dependency. Would it make sense to have an option in Vega to set the locale? Something like `vega.locale('nl-NL')`.

Apparently this API already exists: https://vega.github.io/vega/docs/api/locale/

Is there some plans to implement it? I would also like to be able to set the locale with altair in the end (both in jupyterlab, but also when saving an html).

**Comment by domoritz** [comment]
Great, this API should work. One issue I see now is that we set the locale globally. I wonder whether people would think that the locale is set per spec if it’s passed as an argument to embed when in fact it’s a global setting?

**Comment by victornoel** [comment]
> Great, this API should work. One issue I see now is that we set the locale globally. I wonder whether people would think that the locale is set per spec if it’s passed as an argument to embed when in fact it’s a global setting?

From my experience with d3, I don't think it's going to be possible to scope those settings, but I may be missing something though...

**Comment by domoritz** [comment]
Then we need to think about how this property is exposed in the api. What do you think?

**Comment by victornoel** [comment]
@domoritz maybe just delegate the responsibility of handling this to vega itself by considering the following reasoning:
- vega exposes methods to change its instance locales
- vega-embed just uses those methods, and don't "know" that actually d3 is globally scoped

It's a bit cheating instead of dealing with the problem, but at the same time, think that where this problem can be fixed is at the level of vega itself when it uses d3.

**Comment by domoritz** [comment]
Fixed and released in 6.1.0.
